add email as Braze event streaming identifier option (#120)
amplitude/amplitude-dev-center: docs/data/destinations/braze.md

    - [**External ID**](https://www.braze.com/docs/api/basics/#user-ids): Any unique identifier for each user in Braze.
    - [**Braze ID**](https://www.braze.com/docs/api/basics/#user-ids): A unique identifier provided by Braze for each user.
    - [**User Alias**](https://www.braze.com/docs/api/objects_filters/user_alias_object): An alternative unique identifier for each user in Braze.

>     - [**Email**](https://www.braze.com/docs/api/endpoints/user_data/post_user_track/#frequently-asked-questions): Using email as the identifier without an External ID or Braze ID can lead to some unexpected behaviors. Check the [Braze Documentation](https://www.braze.com/docs/api/endpoints/user_data/post_user_track/#frequently-asked-questions) for more details.

Reviewer: ```suggestion
    - [**Email**](https://www.braze.com/docs/api/endpoints/user_data/post_user_track/#frequently-asked-questions): Using email as the identifier without an External ID or Braze ID may lead to unexpected behavior. 
```

Author: Ah good point thanks!
